resolve https://github.com/neuronets/trained-models/issues/87

hvgazula opened the pull request.

hvgazula commented:
Please look into mine. It came earlier and is cleaner than yours. Thanks.

gaiborjosue commented:
Hello @hvgazula what happened to the workflow "change_label"?

gaiborjosue commented:
Hello @hvgazula the problem with your PR is that it does not consider the fact that the workflow can fail. If failed, the issue will be tagged as "failed" thus the user can't change the label manually to "Ready-to-test" since they are outside collaborators. Since you removed the file that does this automatically, there is no way of re-testing the workflow, it will just stay as "failed"

hvgazula commented:
I am confused. Then what is this- You are asking the user to change the label. How is this possible? https://github.com/neuronets/trained-models/blob/b49f52df7c248295f6ad0205e4bcc568afbaae3d/.github/workflows/new_model.yml#L603-L607

gaiborjosue commented:
Yes sorrry. I forgot to update the bot comment, however in the docs it does say to create an issue comment instead of changing the label.

hvgazula commented:
Sure. Then what exactly is the assingIssue.yml doing? My understanding is as follows:

1. Create issue using template ("ready-to-test" label gets added at issue creation)
2. the issue gets assigned to the creator (this means the user/creator should be able to change labels). Am I right?
3. If workflow fails, the label changes to "failed" (this is probably why the change_label.yml is needed)
4. Once the user addresses the issues, they change the label to "ready-to-test" manually.
5. workflow runs fine..label changes to "success"

gaiborjosue commented:
No. They can't change labels manually.

gaiborjosue reviewed (approved):
Looks good, thanks for fixing the issue!
